scripts/train/train.py

```diff
@@ -427,9 +425,7 @@ def main(cfg: DictConfig) -> Trainer:
             if eval_gauntlet_callback is not None:
                 callbacks.append(eval_gauntlet_callback)
         except BaseContextualError as e:
-            if mosaicml_logger is not None:
-                e.location = EvalDataLoaderLocation
-                mosaicml_logger.log_exception(e)
+            e.location = EvalDataLoaderLocation
```

Comment:
why does location need to be set if the logger is being migrated off of? should we be removing this fn from logger in composer?

Reply:
The location attribute is on the exception class helps us tell the user if their data validation failed for their eval dataset or their train dataset. We still need this attribute set even when we are catching exceptions without the logger.